packages/api/src/task/scheduler.ts

```diff
@@ -165,13 +181,19 @@ export class TaskScheduler {
     return true;
   }
 
-  private async failTask(task: Task, error: string, output?: Task["output"]) {
+  private async failTask(
+    task: WithID<Task>,
+    error: string,
+    output?: Task["output"]
+  ) {
     const status = {
+      ...task.status,
       phase: "failed",
       updatedAt: Date.now(),
       errorMessage: error,
     } as const;
     await this.updateTask(task, {
+      ...task.status,
```

Comment:
Just noticed this... Every failed task has a buggy schema now, with the status inlined in the task object hahah